gbfs-validator/cli.js

```diff
@@ -1,17 +1,65 @@
 const { inspect } = require('util')
+const commander = require('commander')
+const fs = require('fs')
+const fsPath = require('path')
 const GBFS = require('./')
+const pjson = require('./package.json');
 
-if (!process.argv[2]) {
-  console.error('Usage: gbfs-validator GBFS_URL')
-  process.exit(1)
+getFeedValidationReport = async (url) => {
+  const gbfs = new GBFS(url)
+  return gbfs.validation()
 }
 
-const gbfs = new GBFS(process.argv[2])
+parseOptions = () => {
+  commander
+    .version(pjson.version, '-v, --version')
+    .usage('[OPTIONS]...')
+    .option('-u, --url <dataset_folder>', 'URL of the GBFS feed')
+    .option('-p, --print', 'Print report to console')
+    .option('-vb, --verbose', 'Verbose mode prints debugging console logs')
+    .option('-s, --save-report <report_path>', 'Local path to output report file')
+    .parse(process.argv)
 
-if (require.main === module) {
-  gbfs
-    .validation()
-    .then(result => console.log(inspect(result, { depth: null, colors: true })))
-} else {
-  module.exports = gbfs.validation()
+  return commander.opts()
+}
+
+const options = parseOptions();
+
+const saveReport = (report, filePath) => { 
+  const dirname = fsPath.dirname(filePath);
+  if (!fs.existsSync(dirname)) {
+    fs.mkdirSync(dirname, { recursive: true });
+  }
+  fs.writeFileSync(filePath, JSON.stringify(report))
 }
+
+const processFeedValidation = async () => {
+  if (options.verbose) {
+    console.log("Started GBFS validation with options: \n " + inspect(options, { depth: null, colors: true }))
+  }
+  try {
+    const report = await getFeedValidationReport(options.url)
+    if (options.print) {
```

Comment:
Maybe make it default ?
If no `-p` / `-s` arguments is passed, nothing happen, and we don't have any clue

Reply:
What do you think changing `-p` to `-nstdout` or similar as _No standard output_, in addition to trigger an error if `-nstdout` is passed and `-s` is not present.

Reply:
Alternatively, what do you think about removing the `-p` option completely and only print when option `-s` is missing?

Reply:
That's a good solution too. My only concern is to drive the use of the command line too much. Having separate options give more control over what goes to a file and stdout.